### ISSUE TITLE
COMException reading WebResourceResponse content

### ISSUE THREAD

**Issue by GunArm** [MicrosoftEdge/WebView2Feedback#1826]
This is on the line between a question and a bug report.  I'll start with a question.

I have a need in my application to access the document source.  I subscribe to CoreWebView2.WebResourceResponseReceived and handle it like this to populate a documentText string
```
        private async void CoreWebView2_WebResourceResponseReceived(object sender, CoreWebView2WebResourceResponseReceivedEventArgs e)
        {
            string expectedUrl = recentNavigatingUrl?.Split("#")[0];
            if (e.Request.Uri != expectedUrl) return;

            var stream = await e.Response.GetContentAsync();
            documentText = new StreamReader(stream).ReadToEnd();
        }
```

This has worked very reliably until recently we started seeing an issue which only (but reliably occurs) when making requests to specific servers on aws.  On those specific servers, the page loads successfully and renders correctly in the WV2 instance, but an exception is thrown when awaiting the GetContentAsync() method.

`COMException: The pipe is being closed. (0x800700E8)`

This happens on every machine which is connected to those specific servers, but not on the same machines pointed to any other servers.  It is surprising to me that such a failure pattern could occur, where the requests are successful and the content displayed (which would seem to rule out any server related issues), yet reliably this exception awaiting the response content happens every time on these specific servers only.  I assume there are some other confounding factors in play that I am not aware of yet, but they are unknowns.

Can help me understand what this message is, what it could mean?  My naive assumption it is some problem under the hood between the c# nuget webview2 programming interface, and the webview2 runtime, with some kind of (named pipe?) IPC between them closing before it is supposed to, although I can't imagine how that could be server dependent since it would be a failure between two things on the local machine.   Yet this pattern of behavior is very consistent and a big problem for us.

I'm using Microsoft.Web.WebView2 (1.0.992.28) and runtime 94.0.992.38.  Although I've seen it on a variety of versions of each as we have been continually trying to update things to make the problem go away.

Any information you can provide as to what the message might mean, or any info about the architecture between Microsoft.Web.WebView2 and the runtime, from which I might be able to make inferences or come up with ideas, would be helpful.  Or if there is any other way to get the document text (which is not a js call to "document.documentElement.outerHTML;" since that is not the literal response from the server) that could also be helpful.

**Comment by Viajaz** [comment]
> After continued research, closing to open bug without the mostly incorrect baggage from this issue.

Did you find a way to mitigate this issue?

**Comment by GunArm** [comment]
The bug issue I opened was #1839.  It has a less clouded description of what was going on after I narrowed things down.
In the last month it was marked fixed and closed, although I haven't gotten around to re-examining the issue and/or incorporating the fix.
Hard to believe this was only 7 months ago.  Unfortunately, I can't recall without serious digging what if anything I ended up doing to work around it.
